azurerm/resource_arm_kubernetes_cluster_test.go

```diff
@@ -410,7 +441,9 @@ resource "azurerm_kubernetes_cluster" "test" {
 `, rInt, location, rInt, rInt, rInt, clientId, clientSecret)
 }
 
-func testAccAzureRMKubernetesCluster_advancedNetworkingKubenet(rInt int, clientId string, clientSecret string, location string) string {
+func testAccAzureRMKubernetesCluster_advancedNetworking(rInt int, clientId string, clientSecret string, location string,
```

Comment:
given the values are going to be the same for the `dnsServiceIp `, `dockerBridgeCidr ` and `serviceCidr` fields in both Kubenet and Azure in both cases - we can duplicate the test configuration and hard-code the values for both cases?

Reply:
Thanks for the suggestion.The extra parameters provide functionalities to add more testing combinations, besides all empty or all set. I would like to leave it here.

---
In reply to: [207946292](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r207946292) [](ancestors = 207946292)

Reply:
@metacpp this should be refactored to match the other resources, since the values intentionally shouldn't be set to empty strings if they're unset

Reply:
I did the refactoring to align with your preference here.

---
In reply to: [208550336](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r208550336) [](ancestors = 208550336)